### ISSUE TITLE
Preview of a reference to a document with a field named 'title' does not display correctly

### ISSUE THREAD

**Issue by frederikgdl** [sanity-io/sanity#2995]
**Describe the bug**

When I reference a document with a field named `title` the Preview/List preview does not work as expected. It seems to be stringified when selected, making me unable to access nested properties. This does not seems to be the case if the field is named something else. My title is localized and this causes the preview to show 'Untitled' because it fails to access the properties of the string (should be object as before). It still displays correctly in the deskStructure This started with version 2.23.0

**To Reproduce**

1. Have the following in schema:
```
const localizedString = {
  name: 'localizedString',
  title: 'Localized string',
  type: 'object',
  fields: [
    {title: 'Norwegian', name: 'no', type: 'string'},
    {title: 'English', name: 'en', type: 'string'},
  ],
}

const inner = {
  title: 'Inner',
  name: 'inner',
  type: 'document',
  fields: [
    {
      name: 'title',
      type: 'localizedString',
    },
  ],
  preview: {
    select: {
      title: 'title.no',
    },
  },
}

const outer = {
  title: 'Outer',
  name: 'outer',
  type: 'document',
  fields: [
    {
      name: 'innerRef',
      type: 'reference',
      to: [{type: 'inner'}]
    },
  ],
}
```

2. Create a document of type inner
3. When attempting to reference the inner from the outer it displays as 'Untitled'
4. If you add a prepare-function to preview you can see that title is stringified if the select is changed to `title: 'title'`
5. If the `title` field is just a normal string it is still stringified.
6. Does not seem to happen if I name the field something other than `title`

**Expected behavior**

Expect the preview to show the title correctly.

**Screenshots**
<img width="601" alt="Screenshot 2021-12-09 at 11 13 14" src="https://user-images.githubusercontent.com/8751346/145377290-38eecadc-420d-4987-b34e-1af72adc0af5.png">

Here I have added 
```
preview: {
    select: {
      title: 'title',
    },
    prepare({title}) {
      return {
        title: JSON.stringify(title)
      }
    }
  },
```
to the inner document to show it is stringified.

**Which versions of Sanity are you using?**

@sanity/cli                 2.23.0 (up to date)
@sanity/base                2.23.0 (up to date)
@sanity/components          2.14.0 (up to date)
@sanity/core                2.23.0 (up to date)
@sanity/default-layout      2.23.0 (up to date)
@sanity/default-login       2.23.0 (up to date)
@sanity/desk-tool           2.23.0 (up to date)
@sanity/production-preview  2.15.0 (up to date)
@sanity/vision              2.23.0 (up to date)

This works correctly with the following versions:

@sanity/cli                 2.23.0 (up to date)
@sanity/base                2.22.3 (latest: 2.23.0)
@sanity/components          2.14.0 (up to date)
@sanity/core                2.22.3 (latest: 2.23.0)
@sanity/default-layout      2.22.5 (latest: 2.23.0)
@sanity/default-login       2.22.3 (latest: 2.23.0)
@sanity/desk-tool           2.22.5 (latest: 2.23.0)
@sanity/production-preview  2.15.0 (up to date)
@sanity/vision              2.22.3 (latest: 2.23.0)

**What operating system are you using?**
macOS Monterey Version 12.0.1

**Which versions of Node.js / npm are you running?**
8.1.3
v16.13.0

**Comment by ehuelsda** [comment]
I have the same issue. 

In addition there seems to an issue with custom preview components for references. We have a list of references using a custom preview component where the component is completely ignored
```
...,
type: "array",
of: [
  {
      type: "reference",
      preview: {
        component: CustomPreviewComponent
      },
      to: [{ type: "..." }, { type: "..." }]
   }
]
```

**Comment by rowlandwilliams** [comment]
Seem to be having a similar issue.

Heres my schema:

```
const vesselCity = {
  name: "vesselCity",
  title: "Vessel City",
  type: "document",
  fields: [
    {
      name: "cityName",
      title: "City Name",
      type: "string",
    },
    {
      name: "cityCountry",
      title: "City Country",
      type: "string",
    },
  ],
};

const vesselCityColumn = {
  name: "vesselCityColumn",
  title: "Vessel City Coumn",
  type: "object",
  fields: [
    {
      name: "city",
      title: "City Name",
      type: "reference",
      to: { type: "vesselCity" },
    },
    {
      name: "cityVessels",
      title: "Vessels at the City",
      type: "array",
      of: [{ type: "vessel" }],
    },
  ],
  preview: {
    select: {
      title: "city",
    },
    prepare(selection) {
      const { title } = selection;
      console.log(selection);
      return {
        title: title.cityName,
      };
    },
  },
};

 const vesselsTable = {
  name: "vesselsTable",
  title: "Vessels table",
  type: "object",
  fields: [
    {
      name: "vesselTable",
      title: "Vessel Table Columns",
      description:
        "Create a new column for each city in the table.",
      type: "array",
      of: [{ type: "vesselCityColumn" }],
    },
  ],
};
```

Logging selection returns

returns 
```
{title: undefined}
```

And when attempting to preview a column in vesselsTable I get:
![Screenshot 2022-03-11 at 13 23 27](https://user-images.githubusercontent.com/61734284/157875502-166ae829-903b-4866-b6b3-d0176c9b1b37.png)

Am I missing anything here?
Would be very helpful to be able to extract keys from referenced document / objects to show in previews

**Comment by frederikgdl** [comment]
Fixed in [v2.23.1](https://github.com/sanity-io/sanity/releases/tag/v2.23.1)
